.github/workflows/github-pages.yaml

```diff
@@ -1,25 +1,31 @@
 name: GitHub Pages Deploy
+
 on:
   release:
     types: [published, prereleased]
   workflow_dispatch:
+
 permissions:
   contents: write
+
 jobs:
   deploy:
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v4
+      - name: Checkout Repository
+        uses: actions/checkout@v4
         with:
-          python-version: 3.x
-      - uses: actions/cache@v4
+          fetch-depth: '0'  # Fetch all commit history for all branches as well as tags.
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
```

Comment:
Reason behind upgrading `actions/setup-python` to `v5` is pretty simple.. it just works after upgrading this action, and I can confirm this after testing it in a new branch.

I've branched-out & committed/pushed a [very simple commit](https://github.com/og-mrk/winutil/commit/c409409a0dc1a0a2d662710170cb5fcb7fc2d036), which even after deleting the branch, still exists thanks to GitHub having everything stay even after "Deleting" it, which's an awesome feature for some people.. _but somewhat terrifying to others_, but nonetheless.. it does work & produces no errors/fails to build & deploy the docs website, [link to the run](https://github.com/og-mrk/winutil/actions/runs/10302979965).